### PR TITLE
Provide overdue search history string set from preferences

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.shortcut.com/simpledotorg/story/8520/display-search-history-in-overdue-search-screen

### CODE REVIEW

**Comment by msasikanth** [comment]
Current dependencies on/for this PR:
* master
  * **PR #3951** <a href="https://app.graphite.dev/github/pr/simpledotorg/simple-android/3951" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/simpledotorg/simple-android/3951?utm_source=stack-comment).

**Review by janhavisinghh — APPROVED** [review]

**Review by janhavisinghh — APPROVED** [review]

**Comment by msasikanth** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/simpledotorg/simple-android/3951?utm_source=stack-comment).


Job Reference: QO6BNrFvMbihWnEksjTV

**Pull request merged**
